app/validation/validator.py

```diff
@@ -497,48 +496,65 @@ def _validate_answer_ids_present_in_schema(self, when, answer_ids_with_group_id,
 
         return errors
 
-    def _validate_condition_in_when_rule(self, when, answer_ids_with_group_id):
+    def _validate_checkbox_exclusive_conditions_in_when_rule(self, when, answer_ids_with_group_id):
         """
-        Validate when rule with 'type' key is correct
+        Validate checkbox exclusive conditions are only used when answer type is Checkbox
         :return: list of dictionaries containing error messages, otherwise it returns an empty list
         """
         errors = []
 
         condition = when['condition']
-        checkbox_exclusive_conditions = ('contains any', 'contains all')
+        valid_checkbox_conditions = ('contains any', 'contains all', 'contains', 'not contains', 'set', 'not set')
         answer_type = answer_ids_with_group_id[when['id']]['answer']['type'] if 'id' in when else None
 
         if answer_type == 'Checkbox':
-            if condition not in checkbox_exclusive_conditions:
-                errors.append(self._error_message(f'The condition`{condition}` cannot be used with `Checkbox` answer type.'))
-        elif condition in checkbox_exclusive_conditions:
+            if condition not in valid_checkbox_conditions:
+                errors.append(self._error_message(f'The condition `{condition}` cannot be used with `Checkbox` answer type.'))
+        elif condition in ('contains any', 'contains all', 'contains', 'not contains'):
```

Comment:
I know this is how it was originally, but now that there are more conditions in the list I think it would make sense to have:

```
checkbox_exclusive_conditions = ['contains any', 'contains all', 'contains', 'not contains']
all_checkbox_conditions = checkbox_exclusive_conditions + ['set', 'not set']
```

Reply:
Changed back.